app/ide-desktop/eslint.config.js

```diff
@@ -0,0 +1,354 @@
+/** @file ESLint configuration file. */
+/** NOTE: The "Experimental: Use Flat Config" option must be enabled. */
+import * as path from 'node:path'
+import * as url from 'node:url'
+
+const DIR_NAME = path.dirname(url.fileURLToPath(import.meta.url))
+
+// The .d.ts is wrong; these only have default exports.
+// Other default imports are whitelisted; however these should only be used here.
+/* eslint-disable no-restricted-syntax */
+import eslintJs from '@eslint/js'
+import globals from 'globals'
+import jsdoc from 'eslint-plugin-jsdoc'
+import tsEslint from '@typescript-eslint/eslint-plugin'
+import tsEslintParser from '@typescript-eslint/parser'
+/* eslint-enable no-restricted-syntax */
+
+const NAME = 'enso'
+/** An explicit whitelist of CommonJS modules, which do not support namespace imports.
+ * Many of these have incorrect types, so no type error may not mean they support ESM,
+ * and conversely type errors may not mean they don't support ESM -
+ * but we add those to the whitelist anyway otherwise we get type errors.
+ * In particular, `yargs` and `string-length` support ESM but their type definitions don't. */
+const COMMONJS_MODULES =
+    'chalk|string-length|yargs|yargs\\u002Fyargs|sharp|to-ico|connect|morgan|serve-static|create-servers|electron-is-dev|fast-glob|esbuild-plugin-alias|esbuild-plugin-time|esbuild-plugin-yaml'
+const OUR_MODULES = 'enso-content-config'
+const RELATIVE_MODULES =
+    'bin\\u002Fproject-manager|bin\\u002Fserver|config\\u002Fparser|authentication|config|debug|index|ipc|naming|paths|preload|security'
```

Comment:
This looks like something extremely painful to maintain. Is there any other way or we need to list these things explicitly here? :(

Reply:
it might be possible to programmatically list directories and generate this string - would that be preferable?

Reply:
Probably not now. But I'm sad it's so complex and that eslint can't do it by itself.